src/lib.rs

```diff
@@ -186,7 +200,12 @@ pub fn consumer(
                             }
 
                             new_results
+                        };
+
+                        if guess_directory {
```

Comment:
Nit: could we call rename_single_files right before add_results, so we don't have to duplicate this code?

Reply:
no, we can't because we need to have the stem.